addon/components/abstract-input.js

```diff
@@ -102,26 +102,25 @@ export default Component.extend(PropTypeMixin, {
 
   /**
    * This should be overriden by inherited inputs to convert the value to the appropriate format
-   * @param {Boolean|String} value - value to parse
+   * @param {Boolean|String} data - value to parse
    * @returns {any} parsed value
    */
-  parseValue (value) {
-    return value
+  parseValue (data) {
+    return data.value || _.get(data, 'target.value') || data
```

Comment:
This logic appears to have been moved from the `onChange` action and I believe is broken. If `data.value` is `false` for the boolean input then the `||` will evaluate.